backend/src/data/repositories/course/course.repository.ts

```diff
@@ -217,36 +219,32 @@ class Course {
         }
       })
       .whereIn('mentors.id', filteredMentorIds)
-      .select(
-        'mentors.id',
-        'fullName',
-        'gender',
-        'dateOfBirth',
-        'mentors:userDetails:avatar as avatar',
-      )
+      .select('mentors')
       .withGraphJoined(
         'mentors(withoutPassword).[userDetails(withoutMoneyBalance).[avatar]]',
       )
-      .castTo<UserDetailsResponseDto[]>()
-      .execute();
+      .first()
+      .castTo<CourseAllMentorsDto>();
+
+    return mentors?.mentors ?? [];
   }
 
-  public getMenteesByCourseIdAndMentorId({
+  public async getMenteesByCourseIdAndMentorId({
     courseId,
     mentorId,
-  }: CourseGetMenteesByMentorRequestDto): Promise<UserDetailsResponseDto[]> {
-    return this.#CourseModel
+  }: CourseGetMenteesByMentorRequestDto): Promise<UsersGetResponseDto[]> {
+    const mentees = await this.#CourseModel
       .query()
-      .select('mentees.id', 'fullName', 'mentees:userDetails:avatar as avatar')
+      .select('mentees')
       .withGraphJoined(
-        '[mentees.[userDetails(withoutMoneyBalance).[avatar]], mentors]',
+        'mentees(withoutPassword).[userDetails(withoutMoneyBalance).[avatar]]',
       )
-      .where('mentors.id', mentorId)
-      .andWhere('courses.id', courseId)
+      .where({ mentorId, courseId })
       .andWhereNot('status', MenteesToMentorsStatus.COMPLETED)
-      .distinct('mentees.id')
-      .castTo<UserDetailsResponseDto[]>()
-      .execute();
+      .first()
+      .castTo<GetAllMenteesDto>();
```

Comment:
can we leave the old cast type to avoid a new type?

Reply:
i don't think so, previously query result wasn't corresponding with cast type and it was not possible to get avatar
also for me it looks more logical to have mentors and mentees as users and in another queries we actually return them as users and not user details

Reply:
It looks strange now `mentees?.mentees`. at least because of name duplication
can we flat it and just return an array of mentees ?